### PR TITLE
AudioIO: add spectrogram samples for C++/python

### PR DESCRIPTION
**Pull request by SinM9**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by SinM9** [comment]
@allnes Check please

File: samples/cpp/spectrogram.cpp [hunk 1]
```diff
@@ -0,0 +1,1124 @@
...
+        else
+        {
+            cerr << "Grab error" << endl;
+            break;
+        }
+    }
+
+}
+
+static bool checkArgs(CommandLineParser parser)
```
Comment (alalek) [inline samples/cpp/spectrogram.cpp:861]: Input parameters should be passed as 'const reference'

File: samples/cpp/spectrogram.cpp [hunk 2]
```diff
@@ -0,0 +1,1124 @@
...
+    int currentSamples = 0;
+    vector<vector<double>> stft;
+    int xmin = 0;
+    int xmax = 0;
+    while ((sysTimeCurr - sysTimePrev) / cvTickFreq < microTime)
+    {
+        if (cap.grab())
+        {
+            for (int nCh = 0; nCh < numberOfChannels; nCh++)
+            {
```
Comment (alalek) [inline samples/cpp/spectrogram.cpp:794]: There is one imshow for multiple channels.

File: samples/cpp/spectrogram.cpp [hunk 3]
```diff
@@ -0,0 +1,1124 @@
...
+static vector<int> readAudioMicrophone()
+{
+    VideoCapture cap;
+    vector<int> params {    CAP_PROP_AUDIO_STREAM, 0,
+                            CAP_PROP_VIDEO_STREAM, -1   };
+
+    cap.open(0, CAP_MSMF, params);
+    if (!cap.isOpened())
+    {
+        cerr << "ERROR! Can't to open microphone" << endl;
```
Comment (berak) [inline samples/cpp/spectrogram.cpp:64]: better:

    cerr << "ERROR! Can't open microphone" << endl;

(w/o the "to")

**Review by alalek — APPROVED** [review]
Well done! Thank you :+1:

**Pull request merged**